File: app/src/main/java/com/shmedo/mcloudapp/deviceconfig/model/PageInfo.java
package com.shmedo.mcloudapp.deviceconfig.model;

/**
 * 创建者:   gonghe <br/>
 * 创建时间:  2020/8/21 <br/>
 * 描述：     数据分页信息
 */
public class PageInfo {
    private int page = 0;
    private int firstPage = 0;

    public PageInfo() {
    }

    public PageInfo(int firstPage) {
        this.firstPage = firstPage;
        this.page = firstPage;
    }

    public void nextPage() {
        page++;
    }

    public void reset() {
        page = firstPage;
    }

    public int getPage() {
        return page;
    }

    public boolean isFirstPage() {
        return page == firstPage;
    }
}

File: lib_core/src/main/java/com/shmedo/lib/core/base/model/PageInfo.kt
package com.shmedo.lib.core.base.model

/**
 * 创建者:   gonghe <br></br>
 * 创建时间:  2020/8/21 <br></br>
 * 描述：     数据分页信息
 */
class PageInfo(private var firstPage: Int = 0) {
    var page = firstPage
        private set

    fun nextPage() {
        page++
    }

    fun reset() {
        page = firstPage
    }

    fun isFirstPage(): Boolean {
        return page == firstPage
    }

}
